pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -1286,6 +1381,55 @@ func (m *kubeGenericRuntimeManager) SyncPodContainer(pod *v1.Pod, podStatus *kub
 	return
 }
 
+func (m *kubeGenericRuntimeManager) updateContainerLimits(pod *v1.Pod, podStatus *kubecontainer.PodStatus, containersToUpdate []int) error {
```

Comment:
this seems update all resource limits and for all containers. should it be a atomic operation ? for example, if update failed in the middle for some reason, it will end up some are updated and some are not. what would be the consequence of this case ? retry can help but not guaranteed.

Reply:
this is one of the things we discussed in sig-node earlier this week. short answer: yes it will, and that's ok.

the runtime will report error on partial update, and we don't set our status values to the updated values, and we will keep retrying in order to reconcile. as long as the ordering of pod cgroup sizing update is done with failure cases into account, we will be fine. this would be the most robust approach.

when i add multi-container support, i'll also be splitting up cpu & memory so we won't be updating both at the same time. but we are not telling runtimes to rely on that.